channeldb/db.go

```diff
@@ -240,6 +267,61 @@ func Open(dbPath string, modifiers ...OptionModifier) (*DB, error) {
 	return db, err
 }
 
+func resetChanStateCache(cache *kvdb.Cache) error {
+	cache.Wipe()
+
+	cache.SkipBucket(closedChannelBucket)
+	cache.SkipBucket(closeSummaryBucket)
+	cache.SkipBucket(fwdPackagesKey)
+	cache.SkipBucket(revocationLogBucket)
+
+	if err := cache.AddTopLevelBucket(nodeBucket); err != nil {
+		return err
+	}
+
+	if err := cache.AddTopLevelBucket(edgeBucket); err != nil {
+		return err
+	}
+
+	if err := cache.AddTopLevelBucket(edgeIndexBucket); err != nil {
+		return err
+	}
+
+	if err := cache.AddTopLevelBucket(graphMetaBucket); err != nil {
+		return err
+	}
+
+	// Add other buckets holding channel state to the channeldb cache.
+	if err := cache.AddTopLevelBucket(openChannelBucket); err != nil {
+		return err
+	}
+
+	// Won't fetch.
+	if err := cache.AddTopLevelBucket(closedChannelBucket); err != nil {
```

Comment:
`SkipBucket` just marks a bucket name that we'll skip through when encountered, but we still need to mark it in the cache to be known.

Reply:
So all top level buckets that are accessed through the cached backend need to be pre-registered? Why not do the inverse and only register every bucket that you do want to cache (see earlier comment)? That also prevents future changes from accidentally resulting in OOM because a bucket is silently cached.

Reply:
Did you look at the latest changeset?